Kerberos.NET/Replay/MemoryTicketCache.cs

```diff
@@ -2,95 +2,65 @@
         }
 
-        internal Func<CacheEntry, Task> Refresh { get; set; }
-
-        public bool RefreshTickets { get; set; }
-
-        public TimeSpan RefreshInterval { get; set; } = TimeSpan.FromSeconds(30);
-
-        private async Task RunBackground()
-        {
-            while (true)
-            {
-                if (cts.Token.IsCancellationRequested)
-                {
-                    break;
-                }
-
-                try
-                {
-                    await BackgroundCacheOperation();
-                }
-                catch (Exception ex)
-                {
-                    logger.LogWarning(ex, "Background cache operation failed");
-                }
+        private readonly ConcurrentDictionary<string, CacheEntry> cache
+            = new ConcurrentDictionary<string, CacheEntry>();
 
-                await Task.Delay(RefreshInterval, cts.Token);
-            }
-        }
+        public bool BlockUpdates { get; set; }
 
-        private async Task BackgroundCacheOperation()
+        protected override async Task BackgroundCacheOperation()
         {
-            var originalKeys = cache.Keys.ToArray();
+            var originalKeys = this.cache.Keys.ToArray();
 
             foreach (var kv in originalKeys)
             {
-                if (!cache.TryGetValue(kv, out CacheEntry entry))
+                if (!this.cache.TryGetValue(kv, out CacheEntry entry))
                 {
                     continue;
                 }
 
                 if (entry.IsExpired())
                 {
-                    cache.TryRemove(kv, out _);
+                    this.cache.TryRemove(kv, out _);
                 }
-                else if (RefreshTickets)
+                else if (this.RefreshTickets)
                 {
-                    await Refresh?.Invoke(entry);
+                    await(this.Refresh?.Invoke(entry)).ConfigureAwait(true);
                 }
             }
 
-            var finalKeys = cache.Keys.ToArray();
+            var finalKeys = this.cache.Keys.ToArray();
 
             var newKeys = finalKeys.Except(originalKeys);
             var purgedKeys = originalKeys.Except(finalKeys);
 
             if (newKeys.Any() || purgedKeys.Any())
             {
-                logger.LogDebug("Cache Operation. New: {NewKeys}; Purged: {PurgedKeys}", string.Join("; ", newKeys), string.Join("; ", purgedKeys));
+                this.Logger.LogDebug("Cache Operation. New: {NewKeys}; Purged: {PurgedKeys}", string.Join("; ", newKeys), string.Join("; ", purgedKeys));
             }
         }
 
-        private readonly ConcurrentDictionary<string, CacheEntry> cache
-            = new ConcurrentDictionary<string, CacheEntry>();
+        public override Task<bool> AddAsync(TicketCacheEntry entry)
+        {
+            var added = this.Add(entry);
 
-        public bool BlockUpdates { get; set; }
+            return Task.FromResult(added);
```

Comment:
If we don't use ValueTasks, we should cache these kind of task. Like so:
```c#
private static readonly s_trueTask = Task.FromResult(true);
private static readonly s_falseTask = Task.FromResult(false);

public override Task<bool> AddAsync(TicketCacheEntry entry)
{
    return this.Add(entry) ? s_trueTask : s_falseTask;
}
```
So the (repeated) allocation can be avoided.